app/assets/javascripts/directives/views/historyMenu.ts

```diff
@@ -0,0 +1,184 @@
+import { WebDirective } from '../../types';
+import { WebApplication } from '@/ui_models/application';
+import template from '%/directives/history-menu.pug';
+import { SNItem, ItemHistoryEntry } from '@node_modules/snjs/dist/@types';
+import { PureViewCtrl } from '@/views';
+import { ItemSessionHistory } from 'snjs/dist/@types/services/history/session/item_session_history';
+import { RemoteHistoryList, RemoteHistoryListEntry } from 'snjs/dist/@types/services/history/history_manager';
+import { confirmDialog } from '@/services/alertService';
+
+interface HistoryScope {
+  application: WebApplication
+  item: SNItem
+}
+
+class HistoryMenuCtrl extends PureViewCtrl implements HistoryScope {
+
+  diskEnabled = false
+  autoOptimize = false
+  application!: WebApplication
+  item!: SNItem
+  sessionHistory?: ItemSessionHistory
+  remoteHistory?: RemoteHistoryList
+
+  /* @ngInject */
+  constructor(
+    $timeout: ng.ITimeoutService
+  ) {
+    super($timeout);
+    this.state = {
+      fetchingRemoteHistory: false
+    };
+  }
+  
+  $onInit() {
+    super.$onInit();
+    this.reloadHistory();
+    this.fetchServerHistory();
+    this.diskEnabled = this.application.historyManager!.isDiskEnabled();
+    this.autoOptimize = this.application.historyManager!.isAutoOptimizeEnabled();
+  }
+
+  reloadHistory() {
+    this.sessionHistory = this.application.historyManager!.sessionHistoryForItem(this.item);
+  }
+
+  get isfetchingRemoteHistory() {
+    return this.state.fetchingRemoteHistory;
+  }
+
+  set fetchingRemoteHistory(value: boolean) {
+    this.setState({
+      fetchingRemoteHistory: value
+    });
+  }
+
+  async fetchServerHistory() {
+    this.fetchingRemoteHistory = true;
+    this.remoteHistory = await this.application.historyManager!.remoteHistoryForItem(this.item)
+      .finally(() => {
+        this.fetchingRemoteHistory = false;
+      });
+  }
+
+  async openSessionRevision(revision: ItemHistoryEntry) {
+    this.application.presentRevisionPreviewModal(
+      revision.payload.uuid, 
+      revision.payload.content
+    );
+  }
+
+  async openRemoteRevision(revision: RemoteHistoryListEntry) {
+    const itemUuid = this.item.uuid;
+    this.fetchingRemoteHistory = true;
+    revision = await this.application.historyManager!.fetchRemoteRevision(itemUuid, revision);
+    this.fetchingRemoteHistory = false;
+    this.application.presentRevisionPreviewModal(
+      revision.payload.uuid, 
+      revision.payload.content
+    );
+  }
+
+  classForRevision(revision: ItemHistoryEntry) {
+    const vector = revision.operationVector();
+    if (vector === 0) {
+      return 'default';
+    } else if (vector === 1) {
+      return 'success';
+    } else if (vector === -1) {
+      return 'danger';
+    }
+  }
+
+  clearItemHistory() {
```

Comment:
Functions for session related history should be delineated as such. `clearItemSessionHistory`, etc. Same for remote.